### DISCUSSION TITLE
Factoring

### DISCUSSION THREAD

**Discussion by siascend** [josdejong/mathjs#2360] [Ideas]
I want to do simple factoring like the following. For the equation:
x^3 + 2xy^2 + x
I want to factor our an x so the answer would be:
x^2 + 2y^2 + 1.

I cannot find a way to do this. I tried evaluate, derivative, and simplify but none of them worked.
I even tried to use divide with (x^3 + 2xy^2 + x)/x but that did not work. 

Can someone help me?

**Comment by josdejong** [comment]
Thanks for your question. I think it would be possible to add another rule to `simplify` so it is capable of simplifying an expression like `x^3 + 2xy^2 + x`. Anyone interested in looking into this? Help would be welcome!

**Comment by gwhitney** [comment]
Also #1732 is potentially relevant here; if `x` were made a symbol in the sense of that issue, then at least the division would work. There are some draft PRs like #2475 that are working toward this. Hopefully someone (maybe even me) will find the time to push something like that over the goal line.
